src/modules/Remote.rb

```diff
@@ -72,38 +71,53 @@ def main
 
     # Checks if remote administration is currently allowed
     def IsEnabled
-      @allow_administration
+      !IsDisabled()
     end
 
     # Checks if remote administration is currently disallowed
     def IsDisabled
-      !IsEnabled()
+      @mode == :disabled
     end
 
-    # Enables remote administration.
+    # Enables remote administration without vnc manager.
     def Enable
-      @allow_administration = true
+      @mode = :xinetd
+
+      nil
+    end
+
+    # Enables remote administration with vnc manager.
+    def EnableVncManager
+      @mode = :vncmanager
 
       nil
     end
 
     # Disables remote administration.
     def Disable
-      @allow_administration = false
+      @mode = :disabled
 
       nil
     end
 
+    def EnabledVncManager
+      @mode == :vncmanager
+    end
+
     # Reset all module data.
     def Reset
       @already_proposed = true
 
       # Bugzilla #135605 - enabling Remote Administration when installing using VNC
-      @allow_administration = Linuxrc.vnc
+      if Linuxrc.vnc
+        Enable()
+      else
+        Disable()
+      end
 
       Builtins.y2milestone(
         "Remote Administration was proposed as: %1",
-        @allow_administration ? "enabled" : "disabled"
+        IsEnabled() ? "enabled" : "disabled"
```

Comment:
We should log @mode directly, to disambiguate which mode is the default for "enabled".